src/kernel/context.cpp

```diff
@@ -26,6 +29,8 @@ Context::Context()
 Context::~Context()
 {
     ECC_Stop();
+    assert(g_context);
```

Comment:
Would it be more safer to move the assertion to the top of the desctuctor's code?